### PR TITLE
Added support for LinkBase in Avalonia resources generator

### LINKED ISSUE #16056: AvaloniaResource inclusion results in erroneous embedding when resources are outside of project folder
### Describe the bug

Given referenced Avalonia resources:
```xml
<ItemGroup>
<AvaloniaResource Include="..\avaresdir\**">
    <!-- LinkBase does not seem to matter - whether it is present or not -->
    <LinkBase>avaresdir</LinkBase>
   </AvaloniaResource>
</ItemGroup>
```

and the dir structure:

```
src\app.csproj
avaresdir\*.png
```

The resources get resolved as `avares://AppProjectName/../avaresdir/resourcefile.png` instead of `avares://AppProjectName/avaresdir/resourcefile.png`. The generated `-AvaloniaResources` resource stores resource paths as `/../avaresdir/resourcefile.png`.

This would all be ok, but it fails at runtime, because regardless how the resource is referenced in xaml (whether as `avares://avaresdir/resourcefile.png` or `avares://AppProjectName/avaresdir/resourcefile.png` or `averes://AppProjectName/../avaresdir/resourcefile.png` or `avaresdir/resourcefile.png` or `/../avaresdir/resourcefile.png`), it complains that file is not found, presumably because the asset loader loads resources as `/avaresdir/resourcefile.png`.

### To Reproduce

Include resources from outside project folder:

```xml
<ItemGroup>
<AvaloniaResource Include="..\avaresdir\**">
    <!-- LinkBase does not seem to matter - whether it is present or not -->
    <LinkBase>avaresdir</LinkBase>
   </AvaloniaResource>
</ItemGroup>
```

### Expected behavior

Regardless of whether the files are physically coming from, they should be properly included/embedded. `LinkBase` should be taken into account when including such resources from external directories.

### Avalonia version

11.1

### OS

Windows

### Additional context

Furthermore there are related issues:

- File Symlinks are not supported - fails at compile time with totally unrelated error message. In other words if the included files, are not real files, but symlinks, it fails entirely, which is pretty disappointing since symlink support is important.
- `<AvaloniaResource>` does not support `Visible` element is when included from external dir will not be shown in Solution Explorer in any case, again just because it is not coming from within the project dir should not make it not appear in solution explorer when `LinkBase` is specified.

### PR DESCRIPTION
**Pull request by Abdesol**
## What does the pull request do?
adds support for `LinkBase` in avalonia resources generator

## What is the current behavior?
Avalonia asset loader is not resolving for `LinkBase`

## What is the updated/expected behavior with this PR?
Using LinkBase to set resources path works

## Fixed issues
Fixes #16056

### CODE REVIEW

**Comment by kekekeks** [comment]
Please, add a test (a resource outside of the test lib with LinkBase + some test that loads it from code should be sufficient).

File: src/Avalonia.Build.Tasks/GenerateAvaloniaResourcesTask.cs [hunk 1]
```diff
@@ -38,9 +38,12 @@ public Source(ITaskItem avaloniaResourceItem, string root)
                 _sourcePath = SPath.Combine(root, relativePath);
                 Size = (int)new FileInfo(_sourcePath).Length;
                 var link = avaloniaResourceItem.GetMetadata("Link");
+                var linkBase = avaloniaResourceItem.GetMetadata("LinkBase");
                 var path = !string.IsNullOrEmpty(link)
                     ? link
-                    : relativePath;
+                    : !string.IsNullOrEmpty(linkBase)
```
Comment (MrJul) [inline src/Avalonia.Build.Tasks/GenerateAvaloniaResourcesTask.cs:44]: This treats `LinkBase` and `Link` the same, which isn't correct. `Link` is a file path whereas `LinkBase` is the path before the directory globs, i.e. a directory path.

File: tests/Avalonia.Build.Tasks.UnitTest/GenerateAvaloniaResourcesTaskTest.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Avalonia.Build.Tasks.UnitTest;
+
+public class GenerateAvaloniaResourcesTaskTest
+{
+    [Fact]
+    public void Does_Support_LinkBase_In_Avalonia_Resources_Generator()
+    {
+        var path = "path/to/resources";
+        var expected = "/" + path;
```
Comment (MrJul) [inline tests/Avalonia.Build.Tasks.UnitTest/GenerateAvaloniaResourcesTaskTest.cs:15]: The added test doesn't check the right thing: the expected path should be `/path/to/resources/root.xml`.

**Review by MrJul — CHANGES_REQUESTED** [review]
The current implementation is wrong, see previous comments.

I don't think we should handle `LinkBase` in the task itself.
Let MSBuild expand the recursive globs from `LinkBase` into `Link` instead.

This is what the SDK is doing. We should probably copy this behavior.

Reference: https://github.com/dotnet/sdk/blob/74dafbfb0c0d311595da6fc61d24ca57e6d3f80d/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets#L104-L107

**Comment by maxkatz6** [comment]
Closing due to inactivity

**Pull request closed without merging**